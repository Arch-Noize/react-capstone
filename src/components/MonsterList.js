import React from 'react';
import { useSelector } from 'react-redux';
import Monster from './Monster';

const Classes = () => {
  const { monsters } = useSelector((state) => state.monster);

  return (
    <>
      <h1>List of classes</h1>
      <div>
        {monsters.map((item) => (
          <Monster
            key={item.slug}
            id={item.slug}
            name={item.name}
          />
        ))}
      </div>
    </>
  );
};

export default Classes;
