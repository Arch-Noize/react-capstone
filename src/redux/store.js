import { configureStore } from '@reduxjs/toolkit';
import monsterReducer from './class/monsterSlice';

export default configureStore({
  reducer: {
    monster: monsterReducer,
  },
});
