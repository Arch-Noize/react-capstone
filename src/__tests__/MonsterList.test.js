import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MonsterList from '../components/MonsterList';

jest.mock('axios', () => ({
  get: jest.fn(),
}));

describe('renders CountryCards component', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MonsterList />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
