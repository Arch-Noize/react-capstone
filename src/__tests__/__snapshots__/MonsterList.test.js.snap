// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CountryCards component matches the snapshot 1`] = `
<div>
  <div
    class="list"
  >
    <div
      class="container row row-cols-2"
    >
      <div
        class="bar"
      >
        Monsters and their CR
      </div>
    </div>
    <div
      class="bar"
      id="more"
    >
      And many more creatures
       
      <a
        href="https://5e.tools/bestiary.html"
      >
        here
      </a>
    </div>
  </div>
</div>
`;
